clientcontroller/evm_consumer.go

```diff
@@ -0,0 +1,109 @@
+package clientcontroller
+
+import (
+	"fmt"
+
+	finalitytypes "github.com/babylonchain/babylon/x/finality/types"
+	fpcfg "github.com/babylonchain/finality-provider/finality-provider/config"
+	"github.com/babylonchain/finality-provider/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/chaincfg"
+	"go.uber.org/zap"
+)
+
+var _ ConsumerController = &EVMConsumerController{}
+
+type EVMConsumerController struct {
+	evmClient *rpc.Client
+	cfg       *fpcfg.EVMConfig
+	btcParams *chaincfg.Params
+	logger    *zap.Logger
+}
+
+func NewEVMConsumerController(
+	evmCfg *fpcfg.EVMConfig,
+	btcParams *chaincfg.Params,
+	logger *zap.Logger,
+) (*EVMConsumerController, error) {
+	ec, err := rpc.Dial(evmCfg.RPCAddr)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create EVM RPC client: %w", err)
+	}
+	return &EVMConsumerController{
+		ec,
+		evmCfg,
+		btcParams,
+		logger,
+	}, nil
+}
+
+// SubmitFinalitySig submits the finality signature
+func (ec *EVMConsumerController) SubmitFinalitySig(fpPk *btcec.PublicKey, blockHeight uint64, blockHash []byte, sig *btcec.ModNScalar) (*types.TxResponse, error) {
```

Comment:
We could use 

```suggestion
func (ec *EVMConsumerController) SubmitFinalitySig(fpPk *btcec.PublicKey, blockHeight uint64, blockHash []byte, sig *bbn.SchnorrEOTSSig) (*types.TxResponse, error) {
```

where `*bbn.SchnorrEOTSSig` is defined [here](https://github.com/babylonchain/babylon/blob/a3169a3339846d7cd09fdf7b7ffe977901dbf089/types/btc_schnorr_eots.go#L11-L12)

Reply:
The interface requires `*btcec.ModNScalar`, doesn't it?

Reply:
Hmm looks like finality provider uses `*btcec.ModNScalar` for finality signatures everywhere. Given that it's not good to enforce every consumer to depend on Babylon's struct definition let's use `*btcec.ModNScalar` then :+1: